azure-client-runtime/src/main/java/com/microsoft/azure/AzureProxy.java

```diff
@@ -133,19 +135,19 @@ protected Object handleAsyncHttpResponse(final HttpRequest httpRequest, Single<H
                     .flatMap(new Func1<HttpResponse, Single<? extends HttpResponse>>() {
                         @Override
                         public Single<? extends HttpResponse> call(HttpResponse httpResponse) {
-                            final OperationStatus<Object> operationStatus = new OperationStatus<>(httpRequest, httpResponse, serializer);
+                            final PollStrategy pollStrategy = createPollStrategy(httpRequest, httpResponse, serializer);
 
                             Single<HttpResponse> result;
-                            if (operationStatus.isDone()) {
+                            if (pollStrategy == null || pollStrategy.isDone()) {
```

Comment:
a null PollStrategy means this is not a long-running operation.

Reply:
Correct.